relayer/src/connection.rs

```diff
@@ -44,37 +43,37 @@ pub struct Connection {
 #[derive(Clone, Debug)]
 pub struct ConnectionSideConfig {
     chain_id: ChainId,
-    connection_id: ConnectionId,
     client_id: ClientId,
+    connection_id: ConnectionId,
 }
 
 impl ConnectionSideConfig {
     pub fn new(
         chain_id: ChainId,
-        connection_id: ConnectionId,
         client_id: ClientId,
+        connection_id: ConnectionId,
     ) -> ConnectionSideConfig {
         Self {
             chain_id,
-            connection_id,
             client_id,
+            connection_id,
         }
     }
 
     pub fn chain_id(&self) -> &ChainId {
         &self.chain_id
     }
 
-    pub fn connection_id(&self) -> &ConnectionId {
-        &self.connection_id
-    }
-
     pub fn client_id(&self) -> &ClientId {
         &self.client_id
     }
 
-    pub fn set_connection_id(&mut self, id: &ConnectionId) {
-        self.connection_id = id.clone()
+    pub fn connection_id(&self) -> &ConnectionId {
+        &self.connection_id
+    }
+
+    pub(crate) fn set_connection_id(&mut self, id: &ConnectionId) {
+        self.connection_id = id.clone();
```

Comment:
Can we rather have this function take an owned value? I'd rather expose the `clone` to the call site to make it more explicit rather than doing it implicitly under to the hood (seems to be a general good practice in the Rust ecosystem, although we don't currently follow it everywhere).

```suggestion
    pub(crate) fn set_connection_id(&mut self, id: ConnectionId) {
        self.connection_id = id;
```

I know that this code was not added in your PR, but it would be nice to improve on this while we're at it :)

Reply:
Thanks, this is a good suggestion. Just pushed a commit addressing this.